client/client.go

```diff
@@ -655,6 +655,11 @@ type Destination interface {
 //   * Metadata cannot be generated for the downloaded data
 //   * Generated metadata does not match local metadata for the given file
 func (c *Client) Download(name string, dest Destination) (err error) {
+	err, _ = c.DownloadWithCustom(name, dest)
+	return err
+}
+
+func (c *Client) DownloadWithCustom(name string, dest Destination) (err error, custom *json.RawMessage) {
```

Comment:
Is there any reason to split this function? Why not just make it the default behaviour?

Reply:
Actually, on second thought, so as not to break backwards-compatibility with existing clients that depend on this API, we should make DownloadWithCustom an entirely new function, as you have, but not have Download call it.